src/pages/VenuePage/VenuePage.tsx

```diff
@@ -74,11 +79,16 @@ const VenuePage = () => {
     return <>This event does not exist</>;
   }
 
-  if (!event || (event.price > 0 && !eventPurchase) || !venue || !users) {
+  if (!event || !venue || !users || !userPurchaseHistoryRequestStatus) {
     return <>Loading...</>;
   }
 
-  if (event.price > 0 && eventPurchaseRequestStatus && !eventPurchase) {
+  if (
+    (event.price > 0 &&
+      userPurchaseHistoryRequestStatus &&
+      !hasUserBoughtTicket) ||
+    isEventFinished
```

Comment:
I would split this last one from the first 3. Event finished is different from "You're not allowed in"